src/events/deposits.upsert.service.ts

```diff
@@ -34,14 +35,21 @@ export class DepositsUpsertService {
         operation.type === BlockOperation.DISCONNECTED;
 
       if (shouldUpsertDeposits) {
-        const results = await this.upsert(operation);
-        for (const result of results) {
-          deposits.push(result);
-        }
+        await this.graphileWorkerService.addJob<UpsertDepositsOperationDto>(
+          GraphileWorkerPattern.UPSERT_DEPOSIT,
+          operation,
+          {
+            jobKey: `upsert_deposit:${operation.block.hash}:${operation.type}`,
+          },
+        );
+
+        const headHash =
```

Comment:
Decoupling updating the headHash from the actual deposit upsert makes me a little bit nervous, as it opens an avenue where if a job fails for any particular reason, it won't get re-processed. I may be making a mountain out of a molehill here, though

Reply:
I think it's low risk, but that's a fair point. Moved the head hash update within the upsert.